tutorials/basic-topics/greedy.md

```diff
@@ -10,8 +10,120 @@ keywords:
 
-<TutorialAuthors names="@TBC"/>
+<TutorialAuthors names="@abhishek-sultaniya"/>
 
 ## Overview
 
-...
\ No newline at end of file
+Greedy is a strategy to solve various problems optimally.A greedy algorithm constructs a solution to the problem by always making a
+choice that looks the best at the moment. A greedy algorithm never takes back
+its choices, but directly constructs the final solution. For this reason, greedy
+algorithms are usually very efficient.
+
+Rules:
+
+1)Always pick the best choice at any step
+
+2)Works on the assumption that by picking a local optimum solution at each step we will reach a global optimum
+
+3)The greedy algorithm never reconsiders its choices
+
+```
+Example 1: You have coins of denomination Rs 10, Rs 5, and Rs 1.Find the minimum number of coins to get a total of 17 Rs.
+
+Solution: 4 coins: 10+5+1+1
+
+Example 2: You have coins of denomination Rs 10, Rs 7, and Rs 1.Find the minimum number of coins to get a total of 16 Rs.
+Solution: 4 coins: 7+7+1+1
+Solution if greedy was used : 7 coins: 10+1+1+1+1+1+1
+```
+## Example 1: [605 -Can Place Flowers](https://leetcode.com/problems/can-place-flowers/)
+
+```
+You have a long flowerbed in which some of the plots are planted, and some are not. However, flowers cannot be planted in adjacent plots.
+
+Given an integer array flowerbed containing 0's and 1's, where 0 means empty and 1 means not empty, and an integer n, return if n new flowers can be planted in the flowerbed without violating the no-adjacent-flowers rule.
+
+Input: flowerbed = [1,0,0,0,1], n = 1
+Output: true
+```
+
+For a greedy solution, we would solve in such a way that we will always have the best choice at every max.
+Our task is to calculate maximum flowers we can plant.
+Its simple that if there are three consecutive zeroes then the middle one will be planted.But if we have to calculate maximum then we will miss 2 side case this way.
+
+Case 1: 001....
+Here intially we have just 2 consecutive zeroes but we can plant at first place.So we will consuder this case too.
+
+Case 2: ........100
+Here at the end we have just 2 consecutive zeroes but we can plant at last place.So we will consuder this case too.
+
+```cpp
+class Solution {
+public:
+	bool canPlaceFlowers(vector<int>& flowerbed, int p) {
+		int n = flowerbed.size();
+    // count variable will calculate max flowers we can plant.
+		int count = 0; 
+		if (flowerbed[0] == 0 && n == 1)
+			count++;
+      // The following will cover case1
+		if (n > 1 && flowerbed[0] == 0 && flowerbed[1] == 0)
+		{
+			count++;
+			flowerbed[0] = 1;
+		}
+    // MID APPROACH TO CHECK 3 CONSECUTIVE ZEROES.
+		for (int i = 1; i < n - 1; i++)
+		{
+			if (flowerbed[i] == 0 && flowerbed[i - 1] == 0 && flowerbed[i + 1] == 0)
+			{
+				flowerbed[i] = 1;
+				count++;
+			}
+		}
+          // The following will cover case2
+		if (n > 2 && flowerbed[n - 2] == 0 && flowerbed[n - 1] == 0)
+			count++;
+
+		if (count >= p)
+			return true;
+		else
+			return false;
```

Comment:
why not just `return count >= p;`?

Reply:
I wrote it so that a beginner could understand it better.
I thought he might be confused if I will write just return count >= p;
For better and fast understand i wrote like above.

Reply:
Use `return count >= p;` instead.